docs/investigations/file-upload-with-graphql.md

```diff
@@ -0,0 +1,133 @@
+# File upload with graphql
+
+## Overview
+
+This document discuss the issue of uploading files with graphql. \
+As vanilla graphql does not support uploading files, there are several workarounds
+
+### File uploading with gqlgen 0.9.0 library
+<details>
+<summary>Example of schema.graphql</summary>
+<p>
+
+```graphql
+# The `Upload` scalar type represents a multipart file upload.
+# It is already implemented in gqlgen library, so we can use it straight away.
+scalar Upload
+
+type Documentation {
+  id: ID!
+  data: Upload!
```

Comment:
Are you sure `Upload` scalar can be a field of queryable type? If so, maybe we should add query that returns this type?